tests/rptest/tests/e2e_iam_role_test.py

```diff
@@ -14,12 +14,13 @@ def __init__(self, test_context, extra_rp_conf=None):
         if not extra_rp_conf:
             extra_rp_conf = {}
 
-        super().__init__(test_context,
-                         extra_rp_conf,
-                         environment={
-                             'RP_SI_CREDS_API_HOST': self.iam_server.hostname,
-                             'RP_SI_CREDS_API_PORT': self.iam_server.port,
-                         })
+        super().__init__(
+            test_context,
+            extra_rp_conf,
+            environment={
+                'RP_SI_CREDS_API_ADDRESS':
+                f'{self.iam_server.hostname}:{self.iam_server.port}',
```

Comment:
nit: perhaps worth adding an `address` member to MockIamRolesServer/HttpServer? Or reusing the `url` member somehow

Reply:
fixed to reuse the url